### PR TITLE
[Issue 15934] Update get started dockerfile steps

### LINKED ISSUE #13855: docker docs - Glosses over the Dockerfile lines
File: [get-started/02_our_app.md](https://docs.docker.com/get-started/02_our_app/)

Hello. I'm new to Docker, having put off learning it for what feels like a decade. But that means I have a noob's perspective, uncorrupted by The Curse of Knowledge.

So, Part 2 of **Get started** under the section titled **Build the app’s container image**, step 1, you have a lovely Dockerfile with six commands and a syntax hint at the top. Step 2 follows but fails to discuss the lines in the Dockerfile.

It explains that we "copied in our application and used yarn ..." but I have no real idea how that maps to the lines in the Dockerfile. I especially don't understand whether the commands in the Dockerfile are run within the sandboxed process, or outside it. For example, `WORKDIR /app` is that basically a `chdir`? And is it run inside the container process? I guess that's not built yet. Perhaps it runs the base image and then runs all these commands inside that? Is the path relative to where the Dockerfile was found?

It feels like the author knows this so instinctively that they fail to mention it. Or perhaps they think it's obvious or maybe even that by explaining the most basic things, it somehow makes them appear less smart. Maybe you don't want to get into the weeds of all this yet, but the titles of parts 3 - 10 don't look like they're going to discuss it. And besides, you've showed me the commands already, so you have to explain it.  

It should also explain why it has its own ENV and COPY commands, instead of relying on `RUN cp ...` or even explain why the whole thing isn't just a bash script.

The whole thing leaves me more confused.

Thanks

Luke

### LINKED ISSUE #15934: Windows explorer show docker file Part 2
Updates to conversation of closed issue #8934 

Proposed changes
Add image of Windows File Explorer that shows a Dockerfile

What Was Done and Why
I could not easily find out how to create a Dockerfile on windows. Many sites include the Docker site mentions using a Dockerfile, but do not say how to create it. I initially thought it was like git (.git), however, I figured out that it is simply Dockerfile.

To make this more clear, I included an image from Windows file explorer showing the folder structure of after all three files are created. This sheds light that a Dockerfile on Windows has no special circumstances to create it.

Related issues (optional)
PR: https://github.com/docker/docs/pull/8934
Fixes # https://github.com/docker/docs/issues/8933
https://github.com/docker/docker.github.io/issues/8933

### PR DESCRIPTION
**Pull request by craig-osterhout**
### Proposed changes

Split create Dockerfile and add Dockerfile instructions into 2 steps.
Update Dockerfile description.
Add commands to create a Dockerfile.
Add tip to click Dockerfile instructions to learn more.

### Related issues 

Fixes #15934
Fixes #13855

### CODE REVIEW

**Comment by craig-osterhout** [comment]
@participator, thank you again for the feedback. Please take a look at the changes and let us know any if additional feedback.

File: get-started/02_our_app.md [hunk 1]
```diff
@@ -52,10 +80,11 @@ see a few flaws in the Dockerfile below. But, don't worry. We'll go over them.
    CMD ["node", "src/index.js"]
    EXPOSE 3000
    ```
+  > **Tip**
```
Comment (jerae-duffin) [inline get-started/02_our_app.md:83]: Note?

File: get-started/02_our_app.md [hunk 2]
```diff
@@ -35,12 +35,40 @@ we have created a ZIP file containing the application.
...
 
-In order to build the application, we need to use a `Dockerfile`. A
-Dockerfile is simply a text-based script of instructions that is used to
-create a container image. If you've created Dockerfiles before, you might
-see a few flaws in the Dockerfile below. But, don't worry. We'll go over them.
+In order to build the application, you'll need to use a `Dockerfile`. A
+Dockerfile is simply a text-based file with no file extension. A Dockerfile contains a script of instructions that are used to create a container image. 
 
-1. In the `app` folder, the same location as the `package.json` file, create a file named `Dockerfile` with the following contents.
+1. In the `app` folder, the same location as the `package.json` file, create a file named `Dockerfile`. You can use the following commands to create a Dockerfile based on your operating system.
```
Comment (jerae-duffin) [inline get-started/02_our_app.md:41]:
```suggestion
1. In the `app` folder, the same location as the `package.json` file, create a file named `Dockerfile`. You can use the following commands below to create a Dockerfile based on your operating system.
```

File: get-started/02_our_app.md [hunk 3]
```diff
@@ -35,12 +35,40 @@ we have created a ZIP file containing the application.
...
+   ```console
+   $ cd /path/to/app
+   $ touch Dockerfile
+   ```
+
+   <hr>
+   </div>
+   <div id="windows" class="tab-pane fade" markdown="1">
+
+   In the Windows Command Prompt, run the following commands.
```
Comment (jerae-duffin) [inline get-started/02_our_app.md:61]:
```suggestion
   In the Windows Command Prompt, run the following commands listed below.
```
```suggestion
   In the Windows Command Prompt, run the following commands listed below.
```

Comment (jerae-duffin) [inline get-started/02_our_app.md:61]:
```suggestion
   In the Windows Command Prompt, run the following commands listed below.
```

**Review by jerae-duffin — COMMENTED** [review]
Few minor suggestions. Otherwise, LGTM!

File: get-started/02_our_app.md (hunk 1 above)
Comment (craig-osterhout) [reply get-started/02_our_app.md]: Thanks @jerae-duffin. Updated in the following commit.

File: get-started/02_our_app.md (hunk 2 above)
Comment (craig-osterhout) [reply get-started/02_our_app.md]: Thanks jerae-duffin. Updated in the following commit.

File: get-started/02_our_app.md (hunk 3 above)
Comment (craig-osterhout) [reply get-started/02_our_app.md]: Thanks jerae-duffin. Updated in the following commit.

**Review by jerae-duffin — APPROVED** [review]
LGTM!

**Comment by participator** [comment]
Nice update!

**Pull request merged**